lib/FrontendTool/LoadedModuleTrace.cpp

```diff
@@ -915,8 +899,9 @@ static void createFineModuleTraceFile(const InputFile &input, ModuleDecl *MD) {
   }
 }
 
-bool swift::emitFineModuleTraceIfNeeded(ModuleDecl *mainModule,
+bool swift::emitFineModuleTraceIfNeeded(CompilerInstance &Instance,
                                         const FrontendOptions &opts) {
+  ModuleDecl *mainModule = Instance.getMainModule();
   // When lazy type checking is enabled, we may end up with a partial AST.
   // Walking on these partial AST completely may expose latent bugs.
   if (mainModule->getASTContext().TypeCheckerOpts.EnableLazyTypecheck)
```

Comment:
Could you also switch this to skipping emitting the trace for emit module actions, instead of querying lazy typechecking?

Reply:
Happy to make that change! Is there a canonical flag I can use from here?

Reply:
A `switch` over the `FrontendOptions::ActionType` for the job is probably the best way to comprehensively handle this.

Reply:
`FrontendOpts.RequestedAction`